### PR TITLE
RUBY-5044 Add Ruby 3 support

### PR DESCRIPTION
**Pull request by comandeo**

### CODE REVIEW

**Comment by comandeo** [comment]
https://jira.mongodb.org/browse/MONGOID-4833
https://jira.mongodb.org/browse/MONGOID-5044

File: gemfiles/standard.rb [hunk 1]
```diff
@@ -21,8 +27,12 @@ def standard_dependencies
     gem 'timecop'
     gem 'rspec-retry'
     gem 'benchmark-ips'
-    gem 'rspec-expectations', '~> 3.7', '>= 3.8.4'
-    gem 'rspec-mocks-diag', '~> 3.0'
+    if RUBY_VERSION.start_with?('2.')
+      # Brind the dependencies back the following is fixed:
```
Comment (p-mongo) [inline gemfiles/standard.rb:31]: Bring

**Review by p-mongo — COMMENTED** [review]
I am still wondering - does Rails 5.1 continue to work on Ruby 2?

**Comment by comandeo** [comment]
> I am still wondering - does Rails 5.1 continue to work on Ruby 2?

Yes, it does. It is not supported anymore, though. That's why we may want to remove it from our list of supported versions.

**Comment by p-mongo** [comment]
I support doing that  for 7.4 but I don't think it should be done in this PR.

File: gemfiles/standard.rb (hunk 1 above)
Comment (p-mongo) [reply gemfiles/standard.rb:31]: Also 

> Bring the dependencies back *after* the following is fixed:

**Comment by comandeo** [comment]
Sure, you're right. This one does not remove support for 5.1 - https://github.com/mongodb/mongoid/pull/5009/files#diff-f776cb607ce9083ed350e5839c8034744e53b0a1da6541533a0e9871138942d5R36

File: .evergreen/config.yml [hunk 2]
```diff
@@ -773,15 +799,29 @@ buildvariants:
...
+- matrix_name: app-tests-ruby-2.7
   matrix_spec:
     driver: current
     ruby: ruby-2.7
     mongodb-version: '4.4'
     topology: standalone
     app-tests: yes
     # Not currently testing rails master
-    rails: ['5.1', '5.2', '6.0', '6.1']
+    rails: ['5.2', '6.0', '6.1']
```
Comment (p-mongo) [inline .evergreen/config.yml:824]: Does this not remove test coverage for 5.1?

File: .evergreen/config.yml [hunk 3]
```diff
@@ -697,45 +723,45 @@ buildvariants:
...
-    rails: '5.2'
-  display_name: "${rails}, ${driver}, ${mongodb-version}"
+    rails: '6.0'
+  display_name: "${ruby}, ${driver}, ${mongodb-version}, ${topology}, ${rails}"
   run_on:
     - rhel70-small
   tasks:
      - name: "test"
-- matrix_name: "rails-6.0"
+- matrix_name: "rails-6.1"
```
Comment (p-mongo) [inline .evergreen/config.yml:758]: This shouldn't be needed either. The default rails should be the most recent one allowed by the gemspec, which currently is 6.1. We have gemfiles for older versions, meaning 6.0 through 5.1.

File: .evergreen/config.yml (hunk 2 above)
Comment (comandeo) [reply .evergreen/config.yml:824]: Oh, good catch, thanks!

File: .evergreen/config.yml (hunk 3 above)
Comment (comandeo) [reply .evergreen/config.yml:758]: I'll bring back gemfile and configuration for 5.1

File: gemfiles/standard.rb (hunk 1 above)
Comment (comandeo) [reply gemfiles/standard.rb]: Fixed

File: .evergreen/config.yml (hunk 2 above)
Comment (comandeo) [reply .evergreen/config.yml]: Fixed

**Review by p-mongo — APPROVED** [review]

**Pull request merged**